<?xml version="1.0"?>
<launch>
  <!-- if show a image_view window subscribed to the generated stream -->
    <arg name="visualize" default="false"/>
    <arg name="graphit" default="true"/>
    <arg name="rosmastername" default="SATELLITE-S50-B"/>
    <arg name="rgbOrFlow" default="rgb"/>
    <arg name="split" default="1"/>
    <arg name="cfw" default="50"/>

    <arg if="$(eval rgbOrFlow =='rgb' )" name="img" default="/df_node/image"/>
    <arg if="$(eval rgbOrFlow =='flow')" name="img" default="/df_node/flow_xy"/>
    <arg if="$(eval rgbOrFlow =='rgb' )" name="color" default="green"/>
    <arg if="$(eval rgbOrFlow =='flow')" name="color" default="blue"/>

    <machine name="tsn_caffe" address="tsn_caffe" user="root" password="ros_ros" env-loader="/temporal-segment-networks/catkin_ws/src/caffe_tsn_ros/env.sh $(arg rosmastername)" ssh-port="22"/>
    <group ns="$(arg rgbOrFlow)$(arg split)">
        <include file="$(find caffe_tsn_ros)/launch/cf_remote.launch">
          <arg name="machine" value="tsn_caffe"/>
          <arg name="img" value="$(arg img)"/>
          <arg name="rgbOrFlow" value="$(arg rgbOrFlow)"/>
          <arg name="split" value="$(arg split)"/>
          <arg name="cfw" value="$(arg cfw)"/>
        </include>
        <!-- I dont think this is working anymore -->
        <node if="$(arg visualize)" name="class_image_view" pkg="image_view" type="image_view">
          <remap from="image" to="/class_overlay_image_raw" />
        </node>
        <node if="$(arg graphit)" name="bar_graph" pkg="mygraphs" type="bar.py" required="true">
          <remap from="/act" to="action_label_dic" />
          <param name="title" value="$(arg rgbOrFlow): $(arg split)"/>
          <param name="bar_color" value="$(arg color)"/>
        </node>

    </group>
</launch>
